guillotina_ratelimit/__init__.py

```diff
@@ -0,0 +1,58 @@
+from guillotina import configure
+
+
+app_settings = {
+    "ratelimit": {
+        "global": None,
+        "state_manager": "redis",
+        "redis_prefix_key": 'ratelimit-'
+    }
+}
+
+
+_service_rate_limits_cache = {}
+
+
+def get_service_ratelimits(method, view_name):
+    if (method, view_name) not in _service_rate_limits_cache:
+        rate_limits = _get_service_ratelimits(method, view_name)
+        # Cache them forever
+        _service_rate_limits_cache[(method, view_name)] = rate_limits
+        return rate_limits
+
+    return _service_rate_limits_cache[(method, view_name)]
+
+
+def _get_service_ratelimits(method, view_name):
```

Comment:
I don't understand why this function is O(2n) which is inefficient. At the end you can get the value from the view itself. A view on Guillotina may depend on the context interfaces so there is no way to skip traversing if you want the correct view. Any other solution pre traversal may be random on different content/view deffinition.

Reply:
This function is only checking at the "configured" rate limits for a given service.

Right now, as a first implementation, we can only configure those once, and the same rates will be applied for each content type the endpoint is called to.

so for instance, if '@myEndpoint' can be called at the container level and at the item level, and we call it twice:

```
 POST /db/guillotina/@myEndpoint
 POST /db/guillotina/itemxx/@myEndpoint
``` 
We will have 2 separate entries in redis for the service rate limit manager. However, the configured limit will be the same for both of them.